plugins/woocommerce/templates/cart/cart.php

```diff
@@ -93,15 +93,15 @@
 							 * @param array $cart_item The product in the cart.
 							 * @param string $cart_item_key Key for the product in the cart.
 							 */
-							echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', $product_name, $cart_item, $cart_item_key ) . '&nbsp;' );
+							echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', $_product->get_name(), $cart_item, $cart_item_key ) . '&nbsp;' );
 						} else {
 							/**
 							 * Filter the product name.
 							 *
 							 * @since 7.8.0
 							 * @param string $product_url URL the product in the cart.
 							 */
-							echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', sprintf( '<a href="%s">%s</a>', esc_url( $product_permalink ), $product_name ), $cart_item, $cart_item_key ) );
+							echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', sprintf( '<a href="%s">%s</a>', esc_url( $product_permalink ), $_product->get_name() ), $cart_item, $cart_item_key ) );
```

Comment:
@rrennick thoughts on this one? why should the filter apply to the `<a>` element only in this case?

Reply:
Good catch. If the link is passed through the text is appended after the link. We should restore this back to the pre-7.8.0 change which is https://github.com/woocommerce/woocommerce/pull/37830/files#diff-d3262e00ef9c075cc1fc7ae1dcaea582a69a9bbf278bf5b35ef73236168e889bL82

Reply:
It looks like that commit is just adding some docblock. The `<a>` element is still filtered there. I limited the filter to only the link text in my PR because that made sense to me, but it's a deviation from how it used to be. 

https://github.com/faisal-alvi/woocommerce/commit/70cbbeebd422b7c592a49442dd31f83e976a9838

Reply:
> I think you are saying you want me to bring that back?

Yes. I realize what is being passed to the filter is inconsistent but that's the way it was prior to the 7.8 PR.